### ISSUE TITLE
certain pragmas revert to original values rather than roundtripping

### ISSUE THREAD

**Issue by twest820** [sqlitebrowser/sqlitebrowser#1518]
Details for the issue
--------------------
#### What did you do?
Change pragmas on Edit Pragmas tab or via SQL.

#### What did you expect to see?
1) Modified values set for pragmas when Pragmas tab's Save button are clicked are persisted to database and appear when closed reopened. The Write Changes button does not enable in this case.

2) Modified values set for pragmas via the Execute SQL tab are persisted to database and appear when closed and reopened. The Write Changes button does not enable in this case either, though, it does enable if a query only pragma is issued rather than a write pragma (e.g. pragma temp_store rather than pragma temp_store = memory).

#### What did you see instead?
Original pragma values under both pathways described above. This affects attempts to specify at least

Auto Vacuum = Incremental
Synchronous = Normal
Temp Store = Memory
WAL Auto Checkpoint = <any value besides 1000>

Setting Journal Mode = WAL roundtrips successfully.

Useful extra information
-------------------------
The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?
Windows 10

#### What is your DB4S version?
3.10.1 

#### Did you also
- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

Review of the SQLite documentation at https://www.sqlite.org/pragma.html suggests that, in the case of pragma auto_vacuum the DB Browser may not be issuing the subsequent vacuum command required to change pragma to or from none in a database with existing tables. In this case, a workaround for this issue is to execute both the pragma change and vacuum commands in the Execute SQL tab.

Some experimentation with the current SQLite nuget (https://www.nuget.org/packages/System.Data.SQLite.Core/) suggests pragma temp_store may be scoped to a specific connection rather than to the database and therefore non-durable. It also appears auto_vacuum and journal_mode are sensitive to whether SQLite's caller includes them within a transaction. The SQLite documentation appears unclear on these points.

**Comment by justinclift** [comment]
As a general thought, it's probably worth having a go with the latest nightly build too:

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/

We've done a fair bit of work in the ~year since our last major release, including improvements to how we set PRAGMAs. :smile:

**Comment by twest820** [comment]
Thanks, Justin. I just tried the latest build and, unfortunately, at least temp_store and wal_autocheckpoint exhibit the same behaviour with it as I saw with 3.10.1.

**Comment by justinclift** [comment]
Thanks @twest820. :smile:

Sounds like we still have more stuff to fix withe PRAGMAs then after all.

**Comment by MKleusberg** [comment]
You're right about calling VACUUM after changing the ```auto_vacuum``` pragma. We did that for ```page_size``` but not for ```auto_vacuum```. I'll change that in a second.

```temp_store``` is about temporary tables and indices and therefore most likely not saved in the database file because it doesn't affect the database at all.

For the ```wal_autocheckpoint``` PRAGMA there is also a C-API function and in [the documentation for that function](https://www.sqlite.org/c3ref/wal_autocheckpoint.html) they say that the value is only saved for the connection but not in the database file.

And finally for ```synchronous``` I couldn't find anything but because it's related to WAL as well I assume it's also per connection.

**Comment by MKleusberg** [comment]
I have made two changes now:
1) Run ```VACUUM``` after changing the ```auto_vacuum``` PRAGMA. This should be obvious :smile: 
2) Store the values of the three other PRAGMAs in the project file. We did this for some other PRAGMAs with the same problem already, so it only makes sense to do it here as well. It's the only thing we can offer you: if you want these PRAGMAs to be persistent between connections, you can now change them, then save all your settings to a project file (File -> Save Project), and then always open that project file instead of the original database file.

Are you ok to download [tomorrow's nightly build](https://nightlies.sqlitebrowser.org/latest/) and see if it's improving things for you, @twest820? :smile:

**Comment by justinclift** [comment]
> maybe the thing to do for clarity would be to add a couple group boxes to the pragmas tab in order to indicate which settings can be persisted in the database and which are saved in the project?

That does sound like a good idea. :smile:

**Comment by SteveHere** [comment]
The problem still persists. Any attempts at changing the PRAGMA settings are not saved to the SQLite DB file.
